internal/db/schema/migrations/oss/postgres/29/01_kms.up.sql

```diff
@@ -0,0 +1,228 @@
+begin;
+
+-- make the required schema changes to adopt
+-- github.com/hashicorp/go-kms-wrapping/extras/kms/v2 
+
+
+create table kms_schema_version(
+    version text not null, -- pk is not required since it's a one row table enforced by the kms_schema_version_one_row idx.
+    create_time wt_timestamp,
+    update_time wt_timestamp
+);
+
+-- this index ensures that there will only ever be one row in the 
+-- table.  see: https://www.postgresql.org/docs/current/indexes-expressional.html
+create unique index kms_schema_version_one_row
+ON kms_schema_version((version is not null)); 
+
+create trigger default_create_time
+before
+insert on kms_schema_version
+  for each row execute procedure default_create_time();
+
+create trigger update_time_column 
+before 
+update on kms_schema_version 
+  for each row execute procedure update_time_column();
+
+insert into kms_schema_version(version) values('v0.0.1');
+
+-- we can use our existing kms_root_key table.
+
+-- we need to add an add additional constraint to the kms_root_key_version
+-- table: 
+-- https://github.com/hashicorp/go-kms-wrapping/blob/c06f9db9380f2c26cad07cf51f7a324d25dd55ba/extras/kms/migrations/postgres/04_keys.up.sql#L29 
+alter table kms_root_key_version
+  add constraint not_empty_key
+  check (
+      length(key) > 0
+    );
+
+
+-- we need to create kms_data_key and kms_data_key_version tables.  we will
+-- intentionally be using the boundary domain types and trigger functions. 
```

Comment:
Why are we using boundary domain types and trigger functions? Won't this make it more difficult to compare this against the sql in kms?

Reply:
I was expecting the exact migrations froim: https://github.com/hashicorp/go-kms-wrapping/tree/c06f9db9380f2c26cad07cf51f7a324d25dd55ba/extras/kms/migrations/postgres to be added here. Then a followup migration to migrate the data.

Instead this seems like we are defining our own (hopefully) equivalent tables. This seems somewhat risky to me, given that the kms package is tested against the tables, triggers, and functions it provides, and not against these.

Reply:
I'm not sure the two steps your outlining would even work, given the need to keep the versions and timestamps the same (define schema, then convert it).  Yes, the tables are equiv and yes we need to ensure that they are the sam.e

The boundary domain types and functions are equivalent, so I decided to reuse them vs creating dups with diff names... happy to discuss further if you feel strongly.

Reply:
Yea, lets discuss this further, its not so much that I feel strongly, its more that:
1. This is different from what my expectation was as I was reviewing the PRs in go-kms, so I was just surprised by it.
2. I think there is risk here, and it would be nice to talk through some of the pros/cons of this vs other approaches, since this will set a pattern for how boundary and go-kms integrate.

Reply:
Happy to catch up when I'm back next week.